### PR TITLE
Desktop icon / Splash screen / Tooltip

### LINKED ISSUE #105: InVEST icon is missing
We can track an icon file that electron-builder will use as the app's desktop icon.

### PR DESCRIPTION
**Pull request by davemfish**
This PR fixes #105 by adding an icon to this repo that is passed to electron builder. 

I also added a splash screen that displays our traditional invest logo until the main window is rendered. 

And I added a tooltip to the Open button since that button isn't really self-explanatory ( it wasn't any clearer with it's former name either).

Emily, do you mind downloading and trying the Mac installer to confirm the icon & splashscreen are working?

### CODE REVIEW

**Comment by emlys** [comment]
@davemfish the icon is working for me but the splashscreen isn't

**Comment by davemfish** [comment]
> @davemfish the icon is working for me but the splashscreen isn't. it looks like it's expecting there to be an `invest-logo.png` in `src/static`?

Forgot to add the file, thank you!

**Comment by davemfish** [comment]
Looks like I might also need to debug the puppet test given the new addition of the splash screen BrowserWindow.

**Comment by davemfish** [comment]
Okay @emlys thanks for your patience. All the checks are passing now. I did have to re-run them a couple times because a few of Mac integration tests were timeing out on macos. Unfortunately that's hard to pin down and reproduce so I guess if it continues to be a problem we'll make an issue for it.

**Comment by emlys** [comment]
@davemfish I tried it out again and the splash screen is showing up! I'm not sure it's quite having the intended effect on mac, though. After the first time, it only takes about 1 second to load. The splash screen image is noticeably not there at first and shows up ~halfway through the loading time. Which again is really short so it's not a problem, but I would expect a splash screen to be there from the start.

**Comment by davemfish** [comment]
> @davemfish I tried it out again and the splash screen is showing up! I'm not sure it's quite having the intended effect on mac, though. After the first time, it only takes about 1 second to load. The splash screen image is noticeably not there at first and shows up ~halfway through the loading time. Which again is really short so it's not a problem, but I would expect a splash screen to be there from the start.

Thanks, @emlys . This sounds more/less the same as on Windows: 
* Click the launcher 
* nothing happens for ~1s
* Splashscreen displays with blank main window behind it
* < 1s later,  main window is done loading and splashscreen disappears

The splashscreen & everything else loads with `app.on('ready')` event, so I assume the first second of nothingness is before that event fires. And I'm not sure there's any way to display anything before that event. 

https://www.electronjs.org/docs/api/app#event-will-finish-launching
Do you think it's worth trying that `will-finish-launching` just for MacOS?

**Comment by emlys** [comment]
@davemfish I think it's fine like this! Maybe eventually we can look into speeding up the launch time overall.

File: src/main.js [hunk 1]
```diff
@@ -74,15 +76,31 @@ const createWindow = async () => {
...
     },
   });
-
   const menubar = Menu.buildFromTemplate(
     menuTemplate(mainWindow, ELECTRON_DEV_MODE)
   );
   Menu.setApplicationMenu(menubar);
 
-  // and load the index.html of the app.
+  splashScreen = new BrowserWindow({
```
Comment (emlys) [inline src/main.js:84]: I wonder if moving this to the very start of this function would make it show up any faster?

File: src/main.js [hunk 2]
```diff
@@ -74,15 +76,31 @@ const createWindow = async () => {
...
   mainWindow.loadURL(`file://${__dirname}/index.html`);
 
+  mainWindow.once('ready-to-show', () => {
+    splashScreen.destroy();
+    // We should be able to hide mainWindow until it's ready,
+    // but there's a bug where a window initialized with { show: false }
+    // will load with invisible elements until it's touched/resized, etc.
+    // https://github.com/electron/electron/issues/27353
+    // So for now, we have the splashScreen over a blank, white mainWindow.
+    // mainWindow.show();
```
Comment (emlys) [inline src/main.js:101]: should this still be commented out?

File: tests/loadbutton.test.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    <LoadButton
+      openInvestModel={() => {}}
+      batchUpdateArgs={() => {}}
+    />
+  );
+
+  const openButton = await findByRole('button', { name: 'Open' });
+  userEvent.hover(openButton);
+  const hoverText = 'Browse to a datastack (.json) or invest logfile (.txt)';
+  expect(await findByText(hoverText)).toBeInTheDocument();
```
Comment (emlys) [inline tests/loadbutton.test.js:19]: maybe also `unhover` and assert that the tooltip goes away?

**Review by emlys — COMMENTED** [review]
Thanks @davemfish! my only other comment is that one time I tried it, the splash screen stayed superimposed on the main window after it loaded. I can't reproduce it now so just something to be aware of in case it happens again.

**Comment by davemfish** [comment]
> Thanks @davemfish! my only other comment is that one time I tried it, the splash screen stayed superimposed on the main window after it loaded. I can't reproduce it now so just something to be aware of in case it happens again.

Yikes. If that continues to be a problem then it's probably not worth having the splash screen at all.

File: src/main.js (hunk 1 above)
Comment (davemfish) [reply src/main.js:84]: Great idea, I think that helped!

File: src/main.js (hunk 2 above)
Comment (davemfish) [reply src/main.js:101]: Yes, but I updated the comment so that's no longer confusing.

File: tests/loadbutton.test.js (hunk 3 above)
Comment (davemfish) [reply tests/loadbutton.test.js:19]: good idea!

**Review by emlys — APPROVED** [review]
Thanks @davemfish!

**Pull request merged**